service/PayTrace/PayTraceServices.xml

```diff
@@ -0,0 +1,214 @@
+
+            <script><![CDATA[
+                    org.moqui.util.RestClient restClient = ec.service.rest().method(org.moqui.util.RestClient.POST)
+                        .addHeader("Authorization", "bearer ${token.token.access_token}")
+                        .addHeader("Content-Type", "application/json").jsonObject(requestMap)
+                    restClient.uri().protocol("https").host(pgpt.transactionUrl).port(443).path(pgpt.apiVersion).path(url).build()
+                    org.moqui.util.RestClient.RestResponse restResponse = restClient.call()
+                    
+                    if (restResponse.statusCode < 200 || restResponse.statusCode >= 500) {
+                        ec.logger.error("PayTrace API response ${restResponse.statusCode}")
+                        ec.message.addMessage("PayTrace API error")
+                        return
+                    }
+                    responseNode = restResponse.jsonObject()
+            ]]></script>
+
+            <set field="responseCode" from="responseNode.success"/>
+            <set field="reasonCode" from="responseNode.response_code"/>
+            <set field="reasonMessage" from="responseNode.status_message"/>
+
+            <if condition="!responseCode || true != responseCode">
+                <log level="warn" message="Error saving PayTrace info for PaymentMethod [${paymentMethodId}]: ${responseString}"/>
+                <service-call name="create#mantle.account.method.PaymentGatewayResponse" out-map="context"
+                        in-map="[paymentGatewayConfigId:paymentGatewayConfigId, paymentOperationEnumId:'PgoStoreCim',
+                            paymentMethodId:paymentMethodId, paymentId:paymentId, transactionDate:ec.user.nowTimestamp,
+                            responseCode:responseCode, reasonCode:reasonCode, reasonMessage:reasonMessage, resultSuccess:'N', resultError:'Y']"/>
+                <return message="Could not save payment method ${paymentMethodId}, message: ${reasonMessage} error: ${responseNode}"/>
+            </if>
+
+            <service-call name="create#mantle.account.method.PaymentGatewayResponse" out-map="context"
+                    in-map="[paymentGatewayConfigId:paymentGatewayConfigId, paymentOperationEnumId:'PgoStoreCim',
+                            paymentMethodId:paymentMethodId,paymentId:paymentId, transactionDate:ec.user.nowTimestamp,
+                            resultSuccess:'Y', resultError:'N', resultBadCardNumber:'N', referenceNum: '']"/>
+
+            <!-- set the gatewayCimId values for Party and PaymentMethod as needed -->
+            <if condition="paymentMethod.gatewayCimId"><then>
+                <!-- have a payment profile, called updateCustomerPaymentProfileRequest, no need to do anything -->
+            </then><else-if condition="party.gatewayCimId">
+                <!-- no payment profile but we have a customer profile, called createCustomerPaymentProfileRequest -->
```

Comment:
these comments are from Authorize.net services, should be removed or updated for Paytrace.